File: day8/day8.rb
#!/usr/bin/env ruby

require "./Input.rb"

def getPuzzleInput(fileName)
    lines = File.readlines(fileName)
end

def main
    input = getPuzzleInput("puzzle-example.txt")
    inputParsed = Input.new(input)
    solutionOne = inputParsed.getNumUniqueOutputs()
    solutionTwo = inputParsed.getSumOfOutputs()
    puts "Solution One: #{solutionOne}"
    puts "Solution Two: #{solutionTwo}"
end

main()

File: day8/day8.py

from classes import signals
from classes import segments

def main():
    signalObj = signals.Signals("puzzle-input.txt")
    sum = 0
    for i in range(0, len(signalObj.inputs)):
        segmentsObj = segments.Segments(signalObj.inputs[i], signalObj.outputs[i])
        sum += segmentsObj.getSumOfOutputs(signalObj.outputs[i])
    print("Part One Solution: " + str(signalObj.getUniqueOutputCount()))
    print("Part Two Solution: " + str(sum))

main()

# 3570 answer is too low